Infra: remove double SHA1 from CI builds

SlySven opened the pull request:
#### Brief overview of PR changes/additions
This PR attempts to fix things by no longer appending the Git SHA1 to the "MUDLET_VERSION_BUILD" environmental variable in the CI/CB script files.

#### Motivation for adding to Mudlet
It was breaking the PTB builds because their version identification was malformed with two copies of the development branch git SHA1 for PTB (and presumably) Release builds. Testing and PR builds had the HEAD SHA1 for the current branch applied first followed by that of the development branch commit that the branch was rebased onto. Given the values concerned the first one (which is now generated internally by the QMake/CMake build process) seems to be the one that we want to keep around. This is so that it can be embedded in a "resource" file rather than passed to each object file that gets compiled. 

#### Other info (issues closed, discussion etc)
Note: there is some cruft in the `CI/travis.set-build-info.sh` file relating to builds on the Travis platform - but seen they abandoned supporting FOSS projects a few years back there is no need for that baggage however, to keep things simple I have updated that fragment even though it is ripe for excision from there.

Hopefully this will close #7158 and #7159.

SlySven committed:
Fix: remove double SHA1 from CI builds

It was breaking the PTB builds because their version identification was
malformed with two copies of the development branch git SHA1 for PTB
(and presumably) Release builds. Testing and PR builds had the HEAD SHA1
for the current branch applied first followed by that of the development
branch commit that the branch was rebased onto. Given the values concerned
the first one (which is now generated internally by the QMake/CMake build
process) seems to be the one that we want to keep now it is done there
(so that it can be embedded in a "resource" file rather than passed to
each object file that gets compiled) this PR attempts to fix things by
no longer appending the Git SHA1 to the "MUDLET_VERSION_BUILD"
environmental variable.

Note: there is some cruft in the `CI/travis.set-build-info.sh` file
relating to builds on the Travis platform - but seen they abandoned
supporting FOSS projects a few years back there is no need for that baggage
however, to keep things simple I have updated that fragment even though it
is ripe for excision from there.

Hopefully this will close #7158 and #7159.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven commented:
Okay that is too much - somehow I have lost the SHA1 from the file name that is uploaded to the Mudlet site - must have cut off something I shouldn't - or there was side-effects that weren't obvious when I did it...

SlySven committed:
Fix: restore Git SHA1 to various places it was used in CI scripts

I.e. when it was included as part of the `MUDLET_VERSION_BUILD` environmental
variable - which the previous commit removed. We now keep it in a separate
environmental variable `BUILD_COMMIT`.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven commented:
> Okay that is too much - somehow I have lost the SHA1 from the file name that is uploaded to the Mudlet site - must have cut off something I shouldn't - or there was side-effects that weren't obvious when I did it...

Yeah, the CI scripts operated on the basis that it was contained within the `MUDLET_VERSION_BUILD` variable - so when I cut it out of that (because instead it is stored inside the application's resource files) they broke. 

0a4cddc puts the short SHA1 into a `BUILD_COMMIT` environmental variable and then all three systems' build scripts have been amended to include that after the `MUDLET_VERSION_BUILD` variable when it is used (except for "Release" builds).

SlySven committed:
Fix: a type in a Linux/MacOS CI/CB script

I forgot the close of a `bash` `if` is `fi` NOT `end`.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Revise: use PR commit SHA instead of commit of development used for Windows

The wrong commit was being used.

However it is possible that the qmake processing to get the SHA to use is
not behaving as expected and is also getting the development branch head
into which the PR is merged to use. This will need further investigation
and will show up as the wrong SHA1 embedded in the resource files for
Windows builds.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Revise: found another place where the Git SHA1 was needing in Windows CI

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

Kebap committed:
Merge branch 'development' into Fix_removeDoubledSHA1FromCIBuilds

vadi2 commented:
/refresh links

vadi2 commented:
The cmake/macos/linux sha1's seem correct, but not the qmake/windows ones where it says:

> Project MESSAGE: Git SHA1:  9ad243f7
Project MESSAGE: Value written to app-build.txt file:  -testing-pr7160-9ad243f7

But it should be using e98699e.

SlySven committed:
Fix: identify correct SHA1 to use on AppVeyor for Windows

There we need to identify the grandparent of the HEAD commit we are
informed about. This needs to go into the qmake project file (since we now
embed the SHA1 into the resources file) as well as the (Windows)
"set-build-info" script. This issue was already handled for the Linux/MacOS
CI/CB processes in the GitHub actions by the line:
```sh
BUILD_COMMIT=$(git rev-parse --short "${GITHUB_SHA}^2")
```
in the corresponding "set-build-info" file. We are using `~2` rather than
`^2` as because (AFAICT) the former always refers to the ancestor in the
"working" branch whereas the later does only when there are merges in the
neighbourhood - and since the `git rev-parse` is always run it will not
produce error messages about a non-existent commit reference when there
isn't one.

This commit does not address the CMake case because it is not used/usable
in that environment for the moment - but if this works I will port the
same process to that project file.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Revise: tweak in qmnake upper-casing & selection of an env. variable

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Revise: revert tweak to set-build-info.ps1 & try provided value in qmake

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Revise: correct accessing qmake lower(...) function

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Revise: clean-up now we seem to have something that works

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Fix: typo (missing '}' in qmake file

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Fix: correct missing '-' between PR number & SHA1 in Windows archive file

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven commented:
Okay, *that* seems to have worked for WIndows:
![image](https://github.com/Mudlet/Mudlet/assets/6163092/50c46164-5044-444a-8c19-3603cd5c4553)

SlySven committed:
Merge branch 'development' into Fix_removeDoubledSHA1FromCIBuilds

SlySven committed:
Polish: simplify some things and also apply them for a Windows CMake build

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Fix: another blooming typo!

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Revise: final (I hope) bit of polishing

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>
